graphqlapi/src/main/java/io/stargate/graphql/graphqlservlet/HTTPAwareContextImpl.java

```diff
@@ -97,4 +111,61 @@ private boolean hasJWTToken() {
   public Optional<DataLoaderRegistry> getDataLoaderRegistry() {
     return Optional.ofNullable(dataLoaderRegistry);
   }
+
+  public BatchContext getBatchContext() {
+    return batchContext;
+  }
+
+  /**
+   * Encapsulates logic to add multiple statements contained in the same operation that need to be
+   * executed in a batch.
+   */
+  public static class BatchContext {
+    private final List<String> statements = new ArrayList<>();
+    private final Object statementsLock = new Object();
```

Comment:
This can probably use the monitor of `BatchStatement` w/o requiring another lock object.

Reply:
I'm not sure what you mean, we are using strings for now on the graphql side at least until #273 .
In the process, I wanted to leave `BatchContext` implementation as free from data access logic as possible.

Reply:
IOW, why can't this used `synchronized` methods instead of a lock object?

Reply:
got it, I'll change the occurrences to a synchronized method.

Reply:
Done!